base/client.jl

```diff
@@ -424,7 +424,7 @@ function run_main_repl(interactive::Bool, quiet::Bool, banner::Symbol, history_f
         invokelatest(REPL_MODULE_REF[]) do REPL
             term_env = get(ENV, "TERM", @static Sys.iswindows() ? "" : "dumb")
             term = REPL.Terminals.TTYTerminal(term_env, stdin, stdout, stderr)
-            banner == :no || REPL.banner(term, short=banner==:short)
+            banner > 0 && REPL.banner(term, banner)
```

Comment:
If auto -1, will this prevent the banner from showing by default?

Reply:
See the logic in `repl_main`. If `--banner=-1|auto` then it becomes 0/10 depending on if the current session is interactive.